src/clarity-angular/dropdown/dropdown-submenu.ts

```diff
@@ -0,0 +1,50 @@
+import {Component, ContentChildren, HostListener, Input, QueryList} from "@angular/core";
+import {DropdownItem} from "./dropdown-item";
+
+@Component({
+    selector: "clr-dropdown-submenu",
+    template: `
+        <ng-content></ng-content>
+    `,
+    host: {
+        "[class.dropdown-submenu]" : "true",
+        "[class.open]" : "open",
+        "[class.right]" : "openDirection === 'right'",
+        "[class.left]" : "openDirection === 'left'"
```

Comment:
I forgot this comment on this: right/left is probably not enough, we need the full positioning power. If your submenu is too long and you're close to the bottom of the screen, you need to be able to have it go up, with the bottom of the link and the sub-menu aligned.